### ISSUE TITLE
hostvars[inventory_hostname]['ansible_env'].SSH_CONNECTION.split(' ')[2] -- Outcomes change depending on "become" setting

### ISSUE THREAD

**Issue by jwhimpel** [ansible/ansible#77481]
### Summary

When I run this task, var contains an ipv4 address:
```- name: main - Determine ipv4 address
  ansible.builtin.debug:
    var: hostvars[inventory_hostname]['ansible_env'].SSH_CONNECTION.split(' ')[2]
```

When I run this task, var says VARIABLE IS NOT DEFINED:
```- name: main - Determine ipv4 address
  ansible.builtin.debug:
    var: hostvars[inventory_hostname]['ansible_env'].SSH_CONNECTION.split(' ')[2]
  become: true
```

### Issue Type

Bug Report

### Component Name

ansible

### Ansible Version

```console
$ ansible --version
ansible [core 2.12.3]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/jwhimpel/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.10/site-packages/ansible
  ansible collection location = /home/jwhimpel/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.10.4 (main, Mar 25 2022, 00:00:00) [GCC 12.0.1 20220308 (Red Hat 12.0.1-0)]
  jinja version = 3.0.3
  libyaml = True
```

### Configuration

```console
$ ansible-config dump --only-changed
<No output produced>
```

### OS / Environment

NAME="Fedora Linux"
VERSION="36 (Server Edition Prerelease)"
ID=fedora
VERSION_ID=36
VERSION_CODENAME=""
PLATFORM_ID="platform:f36"
PRETTY_NAME="Fedora Linux 36 (Server Edition Prerelease)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:36"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f36/system-administrators-guide/"
SUPPORT_URL="https://ask.fedoraproject.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=36
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=36
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Server Edition"
VARIANT_ID=server

### Steps to Reproduce

```yaml (paste below)

```

### Expected Results

I  would expect that an ipv4 address would be output in both instances.

### Actual Results

```console
With "become: true"
Positional arguments: poc.yml
verbosity: 4
connection: smart
timeout: 10
become_method: sudo
tags: ('all',)
inventory: ('/home/jwhimpel/ansible/inventories/testing',)
forks: 5
1 plays in poc.yml

**********************************************
Without "become: true"
Positional arguments: poc.yml
verbosity: 4
connection: smart
timeout: 10
become_method: sudo
tags: ('all',)
inventory: ('/home/jwhimpel/ansible/inventories/testing',)
forks: 5
1 plays in poc.yml
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by ansibot** [comment]
Files identified in the description:
None

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/devel/ISSUE_HELP.md)

**Comment by jwhimpel** [comment]
Output with "become: true":
ok: [testFedora.jlhimpel.net] => {
    "hostvars[inventory_hostname]['ansible_env'].SSH_CONNECTION.split(' ')[2]": "VARIABLE IS NOT DEFINED!"

Output without "become: true":
ok: [testFedora.jlhimpel.net] => {
    "hostvars[inventory_hostname]['ansible_env'].SSH_CONNECTION.split(' ')[2]": "192.168.1.32"

**Comment by bcoca** [comment]
I cannot reproduce
```
- hosts: all
  tasks:
  - ansible.builtin.debug:
      var: hostvars[inventory_hostname]['ansible_env'].SSH_CONNECTION.split(' ')[2]

  - ansible.builtin.debug:
      var: hostvars[inventory_hostname]['ansible_env'].SSH_CONNECTION.split(' ')[2]
    become: true
```
results in 
```
TASK [ansible.builtin.debug] *********************************************************************************************************************************************************************************************
ok: [host1] => {
    "hostvars[inventory_hostname]['ansible_env'].SSH_CONNECTION.split(' ')[2]": "<IP>"
}

TASK [ansible.builtin.debug] *********************************************************************************************************************************************************************************************
ok: [host1] => {
    "hostvars[inventory_hostname]['ansible_env'].SSH_CONNECTION.split(' ')[2]": "<IP>"
}

```

**Comment by jborean93** [comment]
Have you applied become to the play level when gathering facts. That will affect what env vars are gathered as facts.

**Comment by bcoca** [comment]
The post had the setting on the debug. If it is as play gathering time, it makes sense, since 'sudo' itself will remove most environment variables by default (see `Defaults`  and `env_keep` in sudo man page)

**Comment by jwhimpel** [comment]
Solved:  The unexpected result is the result of sudo removing the requested Environment Variables.  In my real-world ansible role, I applied the "become" directive at the playlist level because almost all the tasks required superuser privileges (including the task that needs the IPv4 address).  I guess the proper methodology would be to remove the playlist level "become" directive and then apply the "become" directive only on the tasks that need them (the vast majority of tasks).

Sorry for the errant copy-n-paste showing my example task.  I did indeed set the "become" directive at the playlist level and not the task level.  Again, sorry for the confusion.

Thanks to all who pointed out the root cause of my problem.
